aws-rds-dbsubnetgroup/src/main/java/software/amazon/rds/dbsubnetgroup/CreateHandler.java

```diff
@@ -19,25 +21,51 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-                .then(progress -> {
-                    final ResourceModel model = progress.getResourceModel();
-                    if (StringUtils.isNullOrEmpty(model.getDBSubnetGroupName()))
-                        model.setDBSubnetGroupName(IdentifierUtils.generateResourceIdentifier(request.getLogicalResourceIdentifier(), request.getClientRequestToken(), DB_SUBNET_GROUP_NAME_LENGTH).toLowerCase());
-                    return ProgressEvent.progress(model, progress.getCallbackContext());
-                })
-                .then(progress -> proxy.initiate("rds::create-dbsubnet-group", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
-                        .translateToServiceRequest((resourceModel) -> Translator.createDbSubnetGroupRequest(
-                                resourceModel,
-                                Tagging.mergeTags(request.getSystemTags(), request.getDesiredResourceTags())))
-                        .backoffDelay(CONSTANT)
-                        .makeServiceCall((createDbSubnetGroupRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(createDbSubnetGroupRequest, proxyInvocation.client()::createDBSubnetGroup))
-                        .stabilize(((createDbSubnetGroupRequest, createDbSubnetGroupResponse, proxyInvocation, resourceModel, context) -> isStabilized(resourceModel, proxyInvocation)))
-                        .handleError((awsRequest, exception, client, resourceModel, context) -> Commons.handleException(
-                                ProgressEvent.progress(resourceModel, context),
-                                exception,
-                                DEFAULT_DB_SUBNET_GROUP_ERROR_RULE_SET))
-                        .progress())
+                .then(progress -> setDbSubnetGroupNameIfEmpty(request, progress))
+                .then(progress -> createDbSubnetGroup(proxy, proxyClient, progress, systemTags))
+                .then(progress -> updateTags(proxyClient, progress, Tagging.TagSet.emptySet(), extraTags))
                 .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
     }
+
+    private ProgressEvent<ResourceModel, CallbackContext> createDbSubnetGroup(final AmazonWebServicesClientProxy proxy,
+                                                                              final ProxyClient<RdsClient> proxyClient,
+                                                                              final ProgressEvent<ResourceModel, CallbackContext> progress,
+                                                                              final Tagging.TagSet systemTags) {
+        return proxy.initiate("rds::create-dbsubnet-group", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                .translateToServiceRequest((resourceModel) -> Translator.createDbSubnetGroupRequest(resourceModel, systemTags))
+                .backoffDelay(CONSTANT)
```

Comment:
As we are changing this code, I would suggest switching to `HandlerConfig` (See this example: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/blob/master/aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/ReadHandler.java#L17-L23). It will give you a way to override the config in the test env like: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/blob/master/aws-rds-dbinstance/src/test/java/software/amazon/rds/dbinstance/CreateHandlerTest.java#L85-L90, this should speed up the tests significantly.